api/v1/resourceslice.go

```diff
@@ -19,7 +19,6 @@ type ResourceSlice struct {
 	Status ResourceSliceStatus `json:"status,omitempty"`
 }
 
-// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
```

Comment:
This started breaking finalizer updates in my test environment. I have no idea how this isn't breaking the tests, but for now it just isn't worth spending the time thinking about.